spec-tests/src/test/java/org/candlepin/spec/imports/ImportCleanupSpecTest.java

```diff
@@ -0,0 +1,139 @@
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+
+package org.candlepin.spec.imports;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.candlepin.spec.bootstrap.assertions.JobStatusAssert.assertThatJob;
+
+import org.candlepin.dto.api.client.v1.AsyncJobStatusDTO;
+import org.candlepin.dto.api.client.v1.ConsumerDTO;
+import org.candlepin.dto.api.client.v1.EntitlementDTO;
+import org.candlepin.dto.api.client.v1.OwnerDTO;
+import org.candlepin.dto.api.client.v1.PoolDTO;
+import org.candlepin.dto.api.client.v1.ProductDTO;
+import org.candlepin.invoker.client.ApiException;
+import org.candlepin.spec.bootstrap.assertions.OnlyInStandalone;
+import org.candlepin.spec.bootstrap.client.ApiClient;
+import org.candlepin.spec.bootstrap.client.ApiClients;
+import org.candlepin.spec.bootstrap.client.SpecTest;
+import org.candlepin.spec.bootstrap.data.builder.Consumers;
+import org.candlepin.spec.bootstrap.data.builder.Export;
+import org.candlepin.spec.bootstrap.data.builder.ExportGenerator;
+import org.candlepin.spec.bootstrap.data.builder.Owners;
+import org.candlepin.spec.bootstrap.data.builder.ProductAttributes;
+import org.candlepin.spec.bootstrap.data.builder.Products;
+
+import com.fasterxml.jackson.databind.JsonNode;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.io.File;
+import java.util.List;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+
+@SpecTest
+@OnlyInStandalone
+public class ImportCleanupSpecTest {
+
+    private ApiClient admin;
+    private OwnerDTO owner;
+
+    @BeforeEach
+    public void beforeAll() {
+        admin = ApiClients.admin();
+        owner = admin.owners().createOwner(Owners.random());
+    }
+
+    @Test
+    @SuppressWarnings("checkstyle:indentation")
+    public void shouldRemoveStackDerivedPoolWhenParentEntIsGone() throws InterruptedException {
+        ProductDTO productDc = Products.random()
+            .attributes(List.of(
+                ProductAttributes.Arch.withValue("x86_64"),
+                ProductAttributes.StackingId.withValue("stack-dc")));
+        ProductDTO productVdc = Products.random()
+            .attributes(List.of(
+                ProductAttributes.Arch.withValue("x86_64"),
+                ProductAttributes.StackingId.withValue("stack-vdc"),
+                ProductAttributes.VirtualLimit.withValue("unlimited")))
+            .derivedProduct(productDc);
+        ExportGenerator exportGenerator = generateWith(generator -> generator
+            .withProduct(productDc)
+            .withProduct(productVdc));
+
+        Export export = exportGenerator.export();
+        ConsumerDTO consumer = admin.consumers().createConsumer(Consumers.random(owner));
```

Comment:
This is a bit confusing, creating a new consumer here. Can you move this line down to be just before this new consumer is used (when it binds a pool)?

Reply:
done